File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, int m, vector<int> score)
{
    int answer = 0;
    int i = 0;
    int size = score.size() / m;
    sort(score.rbegin(), score.rend());
    
    while (size--)
    {
        answer += *min_element(score.begin() + i, score.begin() + m + i) * m;
        i += m;
    }
    return answer;
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;

        Arrays.sort(score);
        
        int min = Integer.MAX_VALUE;
        for (int i = score.length - 1, cnt = 0; i >= 0; i--, cnt++) {
            int value = score[i];
            if (value < min) {
                min = value;
            }
            if ((cnt + 1) % m == 0) {
                answer += min * m;
            }
        }
        
        return answer;
    }
}